lib/sycamore/sycamore/connectors/pinecone/pinecone_writer.py

```diff
@@ -55,13 +56,20 @@ def write_many_records(self, records: list["BaseDBWriter.Record"], target_params
         assert _narrow_list_of_pinecone_records(records), f"Found bad records in {records}"
         index = self._client.Index(target_params.index_name)
         async_results = []
+        wait_on_index(self._client, target_params.index_name)
```

Comment:
I would put this in `get_existing_target_params`. When that returns the index should exist (be writeable). That said it's functionally the same place codepath-wise so if you leave it here that's ok too.